pkg/aws/ec2/ec2.go

```diff
@@ -195,8 +195,8 @@ func parseENI(iface *ec2.NetworkInterface, vpcs ipamTypes.VirtualNetworkMap, sub
 	}
 
 	for _, ip := range iface.PrivateIpAddresses {
-		if ip.PrivateIpAddress != nil {
-			eni.Addresses = append(eni.Addresses, *ip.PrivateIpAddress)
+		if ip.PrivateIpAddress != nil && !ip.Primary {
```

Comment:
This will not compile on 1.8. Please use the following:

```suggestion
		if ip.PrivateIpAddress != nil && (ip.Primary == nil || !*ip.Primary) {
```